internal/logger/target/http/http.go

```diff
@@ -309,19 +318,69 @@ func (h *Target) startHTTPLogger(ctx context.Context) {
 		return
 	}
 
+	buf := bytebufferpool.Get()
+	defer bytebufferpool.Put(buf)
+
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
+	enc := json.NewEncoder(buf)
+	batchSize := h.batchSize
+	if batchSize <= 0 {
+		batchSize = 1
+	}
+
+	payloadType := "application/json"
+	if batchSize > 1 {
+		payloadType = ""
+	}
+
+	timer := time.NewTimer(3 * time.Second)
+	defer timer.Stop()
+
+	var nevents int
 	// Send messages until channel is closed.
-	for entry := range logCh {
-		h.logEntry(ctx, entry)
+	for {
+		select {
+		case entry, ok := <-logCh:
+			if !ok {
+				return
+			}
+			atomic.AddInt64(&h.totalMessages, 1)
+			nevents++
+			if err := enc.Encode(&entry); err != nil {
+				atomic.AddInt64(&h.failedMessages, 1)
+				continue
+			}
+			if nevents == batchSize {
+				h.logEntry(ctx, buf.Bytes(), payloadType)
+				buf.Reset()
```

Comment:
Reusing the buffer will use the same slice you are writing = race.

Either send a copy or create a new buffer. You probably want to be pooling the byte slices.

Also note that the buffer size will be "sticky", so if at some point the buffer grows to 10MB it will keep this size forever.